js_modules/dagster-ui/packages/ui-core/src/search/SearchDialog.tsx

```diff
@@ -220,60 +260,78 @@ export const SearchDialog = ({
     }
   };
 
-  return (
-    <>
-      <ShortcutHandler onShortcut={openSearch} shortcutLabel="/" shortcutFilter={shortcutFilter}>
-        <SearchTrigger onClick={openSearch} data-search-trigger="1">
-          <Box flex={{justifyContent: 'space-between', alignItems: 'center'}}>
-            <Box flex={{alignItems: 'center', gap: 4}}>
-              <div
-                style={{
-                  height: '24px',
-                  width: '24px',
-                  display: 'flex',
-                  alignItems: 'center',
-                  justifyContent: 'center',
-                }}
-              >
-                <Icon name="search" color={Colors.navTextHover()} />
-              </div>
-              <div>{searchPlaceholder}</div>
-            </Box>
-            <SlashShortcut>/</SlashShortcut>
-          </Box>
-        </SearchTrigger>
-      </ShortcutHandler>
-      <Overlay
-        backdropProps={{style: {backgroundColor: Colors.dialogBackground()}}}
-        isOpen={shown}
-        onClose={() => dispatch({type: 'hide-dialog'})}
-        transitionDuration={100}
-      >
-        <Container>
-          <SearchBox hasQueryString={!!queryString.length}>
-            <Icon name="search" color={Colors.accentGray()} size={20} />
-            <SearchInput
-              data-search-input="1"
-              autoFocus
-              spellCheck={false}
-              onChange={onChange}
-              onKeyDown={onKeyDown}
-              placeholder="Search assets, jobs, schedules, sensors…"
-              type="text"
-              value={queryString}
-            />
-            {loading ? <Spinner purpose="body-text" /> : null}
-          </SearchBox>
-          <SearchResults
-            highlight={highlight}
-            queryString={queryString}
-            results={renderedResults}
-            onClickResult={onClickResult}
-          />
-        </Container>
-      </Overlay>
-    </>
+  const searchResults = (
+    <SearchResults
+      highlight={highlight}
+      queryString={queryString}
+      results={renderedResults}
+      filterResults={assetFilterResults}
+      onClickResult={onClickResult}
+    />
+  );
+
+  const searchInput = (
+    <SearchBox hasQueryString={!!queryString.length}>
+      <Icon name="search" color={Colors.accentGray()} size={20} />
+      <SearchInput
+        data-search-input="1"
+        autoFocus
+        spellCheck={false}
+        onChange={onChange}
+        onKeyDown={onKeyDown}
+        placeholder={isAssetSearch ? 'Search assets' : 'Search assets, jobs, schedules, sensors…'}
+        type="text"
+        value={queryString}
+      />
+      {loading ? <Spinner purpose="body-text" /> : null}
+    </SearchBox>
   );
+
+  if (displayAsOverlay) {
```

Comment:
I don't think reusing `SearchDialog` with overriding props is quite the right approach. There are some similarities between global search and this UI, but I think the way to approach this is to have separate components that *share* common components and hooks.

That is, I would suggest that `SearchDialog` be left effectively as-is, without additional props or rendering branches, and some of its pieces could be factored out to be used by both (existing) global search and a new asset search component.